lib/nanoc/extra/link_collector.rb

```diff
@@ -33,24 +45,46 @@ def filenames_per_href
       filenames_per_href
     end
 
+    def filenames_per_resource_uri
+      require 'nokogiri'
+      filenames_per_resource_uri = {}
+      @filenames.each do |filename|
+        resource_uris_in_file(filename).each do |resouce_uri|
+          filenames_per_resource_uri[resouce_uri] ||= Set.new
+          filenames_per_resource_uri[resouce_uri] << filename
+        end
+      end
+      filenames_per_resource_uri
+    end
+
     def external_href?(href)
       href =~ %r{^(\/\/|[a-z\-]+:)}
     end
 
-    def hrefs_in_file(filename)
-      hrefs_in_file = Set.new
+    def uris_in_file(filename, tag_names)
```

Comment:
(FYI: you can mark classes/methods/modules/… as not part of the public API using `@api private`.)